src/actions/actions.js

```diff
@@ -293,26 +293,46 @@ module.exports = (router) => {
         }
 
         try {
-          return (new VM({
+          const params = await hook.alter('actionContext', {
+            jsonLogic: util.FormioUtils.jsonLogic,
+            data: req.body.data,
+            form: req.form,
+            query: req.query,
+            util: util.FormioUtils,
+            moment: moment,
+            submission: req.body,
+            previous: req.previousSubmission,
+            execute: false,
+            _
+          }, req);
+
+          let vm = new VM({
             timeout: 500,
-            sandbox: _.cloneDeep(await hook.alter('actionContext', {
-              jsonLogic: util.FormioUtils.jsonLogic,
-              data: req.body.data,
-              form: req.form,
-              query: req.query,
-              util: util.FormioUtils,
-              moment: moment,
-              submission: req.body,
-              previous: req.previousSubmission,
-              execute: false,
-              _
-            }, req)),
+            sandbox: _.cloneDeep({
+              execute: params.execute,
```

Comment:
No need to cloneDeep these. And "execute" should be "false" not "params.execute"